lib/cloudant.js

```diff
@@ -493,14 +499,14 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   };
 
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
+    self.update(model, {id: id}, data, {}, function(err, docs) {
       if (err && err.statusCode !== 404) return cb(err);
-      else if (err && err.statusCode === 404) {
+      else if (err && err.statusCode === 404 || docs.count === 0) {
```

Comment:
Is that even possible?
If the document has been deleted, or never existed to begin with, I'd expect a `404 Not Found` instead of a docs object with a count of zero.

Additionally, is it `count`, or should that be `length`?